cli/compilers/wasm_wrap.js

```diff
@@ -0,0 +1,25 @@
+function base64ToUint8Array(data) {{
+  const binString = window.atob(data);
+  const size = binString.length;
+  const bytes = new Uint8Array(size);
+  for (let i = 0; i < size; i++) {{
+    bytes[i] = binString.charCodeAt(i);
+  }}
+  return bytes;
+}}
+
+const buffer = base64ToUint8Array("{}");
+const compiled = await WebAssembly.compile(buffer);
+
+const imports = new Set(
+  WebAssembly.Module.imports(compiled).map(m => m.module)
+);
+
+const importObject = Object.create(null);
+for (const module of imports) {{
+  importObject[module] = await import(module);
+}}
+
+const instance = new WebAssembly.Instance(compiled, importObject);
+
+export default instance.exports;
```

Comment:
What's cool about this is I think it would even work with `deno bundle`

Reply:
yeah, actually that would simplify what I was going to do, there are consequences which I will detail in the related issue though.